osmoutils/accum/accum.go

```diff
@@ -207,6 +226,13 @@ func (accum AccumulatorObject) RemoveFromPositionCustomAcc(name string, numShare
 	// Update user's position with new number of shares
 	initOrUpdatePosition(accum, customAccumulatorValue, name, oldNumShares.Sub(numSharesToRemove), unclaimedRewards, position.Options)
 
+	// Update total shares in accum (re-fetch accum from state to ensure it's up to date)
+	accum, err = GetAccumulator(accum.store, accum.name)
+	if err != nil {
+		return err
+	}
+	setAccumulator(accum, accum.value, accum.totalShares.Sub(numSharesToRemove))
+
```

Comment:
I believe we do since the caller can accidentally keep calling the method on the same `AccumulatorObject` without pulling the new one with updated `TotalShares` from state (and imo we shouldn't be relying on that happening if we can prevent the bug in the function)